src/EFCore.PG/Storage/Internal/Mapping/NpgsqlStringTypeMapping.cs

```diff
@@ -10,16 +11,58 @@ namespace Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.Mapping
     // TODO: See #357. We should be able to simply use StringTypeMapping but DbParameter.Size isn't managed properly.
     public class NpgsqlStringTypeMapping : StringTypeMapping
     {
-        public NpgsqlStringTypeMapping([NotNull] string storeType, bool unicode = false, int? size = null)
-            : base(storeType, System.Data.DbType.String, unicode, size) {}
+        /// <summary>
+        /// Static <see cref="ValueComparer{T}"/> for fixed-width character types.
+        /// </summary>
+        /// <remarks>
+        /// See: https://www.postgresql.org/docs/current/static/datatype-character.html
+        /// </remarks>
+        [NotNull] static readonly ValueComparer<string> CharacterValueComparer =
+            new ValueComparer<string>(
+                (x, y) => x != null && y != null && x.AsSpan().TrimEnd() == y.AsSpan().TrimEnd(),
+                x => x != null ? x.TrimEnd().GetHashCode() : 0);
 
-        protected NpgsqlStringTypeMapping(RelationalTypeMappingParameters parameters) : base(parameters) {}
+        /// <inheritdoc />
+        public override ValueComparer Comparer { get; }
 
+        /// <inheritdoc />
+        public override ValueComparer KeyComparer { get; }
+
+        public NpgsqlStringTypeMapping(
+            [NotNull] string storeType,
+            DbType dbType = System.Data.DbType.String,
+            bool unicode = false,
+            int? size = null)
+            : base(storeType, dbType, unicode, size)
+        {
+            if (storeType == "character" && dbType == System.Data.DbType.StringFixedLength)
```

Comment:
A special value comparer which mimicks the PostgreSQL behavior seems like a good idea!

however, it seems to me like this should simply be a different class rather than this special check (possibly inheriting from `NpgsqlStringTypeMapping` for the `ConfigureParameter()` common behavior, not sure if it's worth it)... It would be cleaner and isolate the, uh, idiosyncracy of the `character` type in a separate class rather than complicating the general string type mapping class. We can have it in the same file just below.

Note that at the ADO.NET level, `DbType.StringFixedLength` is completely indistinguishable from `DbType.String` - both are simply mapped to `text` - so it's not a good idea to use it as an important distinguishing factor at the EF Core level. If you have two classes you no longer have to worry about it. We may want to think about it at the ADO.NET level, although I haven't seen any complaints and am not sure what the actual expected semantics are...